src/state/state.test.ts

```diff
@@ -27,14 +27,7 @@ const stateMock: State = {
       timestamp: 'something-silly',
     },
   },
-  signedApiUrls: {
-    '31337': {
-      hardhat: {
-        '0xC04575A2773Da9Cd23853A69694e02111b2c4182':
-          'http://127.0.0.1:8090/0xbF3137b0a7574563a23a8fC8badC6537F98197CC',
-      },
-    },
-  },
+  signedApiUrls: ['http://127.0.0.1:8090/0xC04575A2773Da9Cd23853A69694e02111b2c4182'],
```

Comment:
I only took a quick look today, but this doesn't feel right to me. The reason why I stored the `signedApiUrls` as record of chain and provider was to handle potential different return values of the providers. We agreed that we will call the signed APIs returned by all providers across all chains (but each provider can report only a single signed API).

I'll take a proper look tomorrow.

Reply:
> handle potential different return values of the providers.

I protected the same behaviour. I preferred arrays because no where utilizes records grouped by keys `chain` and `provider`. At the end `Object.values(...)` leveraged to get all signedApiUrls, see old implementation [here](https://github.com/api3dao/airseeker-v2/blob/a99a98acb3a8235edb15aadf8c1274e77281c1a1/src/data-fetcher-loop/data-fetcher-loop.ts#L79).

Reply:
Mhh, I still believe this won't work. We call the `runUpdateFeeds` [here](https://github.com/api3dao/airseeker-v2/blob/bdrhn9/issue190/src/update-feeds-loops/update-feeds-loops.ts#L54) for all providers across all chains concurrently.

So the signed API URLs [here](https://github.com/api3dao/airseeker-v2/blob/bdrhn9/issue190/src/update-feeds-loops/update-feeds-loops.ts#L246) come only from the single provider, but it removes the Signed APIs reported by others.

Reply:
You are right I missed that point 😞

Reply:
No worries. That's pretty non-apparent behavior. I think we should cover that by unit test. Something along these lines:
```
  it('persists signed API URLs reported by different providers/chains', async () => {
    const dataFeed = generateActiveDataFeedResponse();
    const beacons = contractsModule.decodeDataFeedDetails(dataFeed.dataFeedDetails)!;
    const decodedUpdateParameters = contractsModule.decodeUpdateParameters(dataFeed.updateParameters);
    const activeDataFeed = {
      ...omit(dataFeed, ['dataFeedDetails', 'beaconValues', 'beaconTimestamps']),
      decodedUpdateParameters,
      beaconsWithData: contractsModule.createBeaconsWithData(beacons, dataFeed.beaconValues, dataFeed.beaconTimestamps),
      decodedDapiName: utilsModule.decodeDapiName(dataFeed.dapiName),
      signedApiUrls: [],
    };
    const testConfig = generateTestConfig();
    jest.spyOn(stateModule, 'getState').mockReturnValue(
      allowPartial<stateModule.State>({
        config: { ...testConfig, signedApiUrls: ['http://config.url'] },
        signedApiUrls: ['http://url-from-chain-reported-by-different-provider'],
      })
    );
    jest.spyOn(logger, 'warn');
    jest.spyOn(logger, 'info');

    // Skip actions other than generating signed api urls.
    jest.spyOn(getUpdatableFeedsModule, 'getUpdatableFeeds').mockReturnValue([]);
    jest.spyOn(submitTransactionModule, 'getDerivedSponsorWallet').mockReturnValue(ethers.Wallet.createRandom());
    jest.spyOn(submitTransactionModule, 'hasSponsorPendingTransaction').mockReturnValue(false);

    const { signedApiUrls } = await updateFeedsLoopsModule.processBatch(
      [activeDataFeed],
      'default-provider',
      new ethers.JsonRpcProvider(),
      '31337',
      123
    );

    expect(signedApiUrls).toHaveLength(2);
    // TODO: This breaks
    expect(signedApiUrls).toStrictEqual([
      'http://config.url/0xc52EeA00154B4fF1EbbF8Ba39FDe37F1AC3B9Fd4',
      'http://url-from-chain-reported-by-different-provider',
    ]);
  });
```